src/main/java/org/opentripplanner/raptor/rangeraptor/path/PathParetoSetComparators.java

```diff
@@ -36,98 +47,52 @@ private PathParetoSetComparators() {}
-        }
-      } else {
-        if (c2Comp != null) {
-          return comparatorArrivalTimeAndRelaxedC1AndC2(relaxC1, c2Comp);
-        } else {
-          return comparatorArrivalTimeAndRelaxedC1(relaxC1);
-        }
-      }
-    }
+    Objects.requireNonNull(timeConfig);
+    Objects.requireNonNull(costConfig);
 
-    if (includeC1) {
-      if (includeTimetable) {
-        if (c2Comp != null) {
-          return comparatorTimetableAndC1AndC2(c2Comp);
-        } else {
-          return comparatorTimetableAndC1();
-        }
-      }
-      if (preferLatestDeparture) {
-        if (c2Comp != null) {
-          return comparatorDepartureTimeAndC1AndC2(c2Comp);
-        } else {
-          return comparatorDepartureTimeAndC1();
-        }
-      }
-      if (c2Comp != null) {
-        return comparatorWithC1AndC2(c2Comp);
-      } else {
-        return comparatorWithC1();
-      }
-    }
-
-    if (includeTimetable) {
-      if (c2Comp != null) {
-        return comparatorTimetableAndC2(c2Comp);
-      } else {
-        return comparatorTimetable();
-      }
-    }
-    if (preferLatestDeparture) {
-      if (c2Comp != null) {
-        return comparatorStandardDepartureTimeAndC2(c2Comp);
-      } else {
-        return comparatorStandardDepartureTime();
-      }
-    }
-    if (c2Comp != null) {
-      return comparatorStandardArrivalTimeAndC2(c2Comp);
-    }
-    return comparatorStandardArrivalTime();
+    return switch (costConfig) {
+      case NONE -> switch (timeConfig) {
+        case USE_TIMETABLE -> comparatorTimetable();
+        case USE_ARRIVAL_TIME -> comparatorStandardArrivalTime();
+        case USE_DEPARTURE_TIME -> comparatorStandardDepartureTime();
+      };
+      case USE_C1 -> switch (timeConfig) {
+        case USE_TIMETABLE -> comparatorTimetableAndC1();
+        case USE_ARRIVAL_TIME -> comparatorArrivalTimeAndC1();
+        case USE_DEPARTURE_TIME -> comparatorDepartureTimeAndC1();
+      };
+      case USE_C1_AND_C2 -> switch (timeConfig) {
+        case USE_TIMETABLE -> comparatorTimetableAndC1AndC2(c2Comp);
+        case USE_ARRIVAL_TIME -> comparatorWithC1AndC2(c2Comp);
+        case USE_DEPARTURE_TIME -> comparatorDepartureTimeAndC1AndC2(c2Comp);
+      };
+      case USE_C1_RELAXED_IF_C2_IS_OPTIMAL -> switch (timeConfig) {
+        case USE_TIMETABLE -> comparatorTimetableAndRelaxedC1IfC2IsOptimal(relaxC1, c2Comp);
+        case USE_ARRIVAL_TIME -> comparatorArrivalTimeAndRelaxedC1IfC2IsOptimal(relaxC1, c2Comp);
+        case USE_DEPARTURE_TIME -> comparatorDepartureTimeAndRelaxedC1IfC2IsOptimal(
+          relaxC1,
+          c2Comp
+        );
+      };
+      case USE_C1_RELAX_DESTINATION -> switch (timeConfig) {
+        case USE_TIMETABLE -> comparatorTimetableAndRelaxedC1(relaxC1);
+        case USE_ARRIVAL_TIME -> comparatorArrivalTimeAndRelaxedC1(relaxC1);
+        case USE_DEPARTURE_TIME -> comparatorDepartureTimeAndRelaxedC1(relaxC1);
+      };
```

Comment:
Yes, and it allows us to restrict the combinations to just the supported configurations - e.g. we do not support `relaxed-c1`, because we do not have a use-case for it. And, later it will be easy to add the support...